### PR TITLE
vamb params & vamb v3.0.3

### PR DESCRIPTION
**Pull request by yanhui09**
Hi @SilasK 

1. I tried the new version, but still couldn't make vamb work using your dependency setting. I'm not sure if you have recently
used it to rebuild a conda environment. At least, it didn't work for me. 
As Jakob has released new `vamb v3.0.3`, it may be a better idea to use this.

2. the `vamb` params is a bit wired.
`-m` indicates min-length contigs for binning
`-minfasta` indicates min bin size to keep. You can find the description here https://github.com/RasmussenLab/vamb/tree/master/workflow

I chose your default values to keep bins > 2MB, which shall be comparable to the bins from `metabat`. Previously you chose `-minfasta 2000`, you will have too many bins belonging to phage and plasmid....

I have tested it and it worked for my data. You can find the details in the PR.

I will test `semibin` and give you updates later.

### CODE REVIEW

File: atlas/rules/semibin.smk [hunk 1]
```diff
@@ -57,7 +57,7 @@ rule semibin_generate_data_multi:
         fasta=rules.combine_contigs.output,
         bams=expand(rules.sort_bam.output, sample=SAMPLES),
     output:
-        expand("Cobinning/SemiBin/{sample}/{files}",
```
Comment (yanhui09) [inline atlas/rules/semibin.smk:60]: @SilasK  While I run this pipeline, semibin create a folders called "samples", and {sample}/{files} is under this folder. the previous output path seem not to find csv files.

**Comment by yanhui09** [comment]
@SilasK  I could accomplish `semibin_train`. I adopted used your updated code, also `:` as sample separator. It seems that you have posted a similar issue for `Semibin` . Any idea about this? 
It crashes with a message as below
```
2021-09-20 17:51:05,379 - Start training from one sample.
2021-09-20 17:51:07,395 - Generate training data of 0:
Traceback (most recent call last):
  File "/home/projects/cu_10168/people/yanhui/database/atlasdb/conda_envs/6f6114972dc6e65986f2d932980699b0/bin/SemiBin", line 10, in <module>
    sys.exit(main())
  File "/home/projects/cu_10168/people/yanhui/database/atlasdb/conda_envs/6f6114972dc6e65986f2d932980699b0/lib/python3.8/site-packages/SemiBin/main.py", line 947, in main
    training(logger, args.contig_fasta, args.bams, args.num_process,
  File "/home/projects/cu_10168/people/yanhui/database/atlasdb/conda_envs/6f6114972dc6e65986f2d932980699b0/lib/python3.8/site-packages/SemiBin/main.py", line 676, in training
    model = train(
  File "/home/projects/cu_10168/people/yanhui/database/atlasdb/conda_envs/6f6114972dc6e65986f2d932980699b0/lib/python3.8/site-packages/SemiBin/semi_supervised_model.py", line 204, in train
    train_input_1.append(train_data_input[mapObj[str(link[0])]])
KeyError: 'D10_16376'
```

File: atlas/rules/semibin.smk (hunk 1 above)
Comment (SilasK) [reply atlas/rules/semibin.smk:60]: I don't know why there is a different behavior on my and your computer. I will test it again on a different one.

**Comment by SilasK** [comment]
> @SilasK I could accomplish `semibin_train`. I adopted used your updated code, also `:` as sample separator. It seems that you have posted a similar issue for `Semibin` . Any idea about this?
> It crashes with a message as below

For me semibin worked at the end. Could it be that the combined contigs are not made with the `:` character? 
Could you check all the input files for the semibin train if they have the expected format. E.g. `data.tsv` starts with `D10_16376` also the fast headers of the individual fastas. However the combined should be `>D10:D10_16376`

In the future, I would stick to the `:`sep character and only keep it as a hidden config option. 
Do you have more than one file?

**Comment by yanhui09** [comment]
> [bioconda/bioconda-recipes#30608](https://github.com/bioconda/bioconda-recipes/pull/30608)
> Once the conda package for vamb 3.03 is build could you adapt the vamb.yaml to `vamb>=3.0.3, <3.1 `I suggest.

I will update that later.

**Comment by SilasK** [comment]
I merged and put v3.0.3 in the env that doesn't yet exits

**Comment by yanhui09** [comment]
https://github.com/metagenome-atlas/atlas/blob/adcd292a6ed3a4f23802e1a8980453099c786bec/atlas/rules/cobinning.smk#L37-L40

1. Why do you need to include the `{sample}/{sample}_contigs.fasta` to create `combined.fasta.gz` ?  Only filtered contigs shall be used as input for semibin, right? `zip` shall aggregate all contigs, while shorter contigs will not have information for semibin binning. I met a error in the bin process for some contigs which are in the `combined.fasta` but not in other list. I'm re-testing it. The `predict_taxonomy` process took much time. 

https://github.com/metagenome-atlas/atlas/blob/4e440fa62281f1f980af9c7f7cf7747fe809cfde/atlas/rules/semibin.smk#L21-L25

2. Also, I don't understand why `{sample}/{sample}_contigs.fasta` is used as a input for nearly all `semibin`rules. Is this used as a check procedure?

3. It seems I couldn't run `semibin` independently. It adopted the output of `combined_fasta` in `cobinning`. But if the output doesn't exist, I couldn't let it trigger the previous rule, i.e., `filter_contigs`.

**Comment by SilasK** [comment]
I think  I just merged the first PR from someone else! :horray

**Pull request merged**